modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionContext.java

```diff
@@ -152,6 +156,7 @@ public ExecutionContext(
         this.txAttributes = txAttributes;
         this.timeZoneId = timeZoneId;
         this.inBufSize = inBufSize < 0 ? Commons.IN_BUFFER_SIZE : inBufSize;
+        this.currentUser = username == null ? Commons.SYSTEM_USER_NAME : username;
```

Comment:
I think, the user must be known by this point, hence from execution point of view it never null. The better place to make decision is somewhere prior to call to queryExecutor. We might expose information about user in SQL_QUERIES view, and user must be unambiguously resolved to avoid discrepancies

Reply:
Done.